### PR TITLE
Upgrading to operator 0.19.0

### PR DESCRIPTION
**Pull request by CharlyF**
### What does this PR do?

- Bumping the operator.
- Main change is on the supported types, from strings to anyOf.
-  Enforce tolerance and scaling windows values in the code (can't use kubebuilder's features).
- Improve logging.

### Motivation

step update to get the operator 1.0.0.

### Describe your test plan

Update controller with existing WPAs and create new ones, make sure nothing is mutated and results are expected.

### CODE REVIEW

File: deploy/olm-catalog/watermarkpodautoscaler/manifests/watermarkpodautoscaler.clusterserviceversion.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@ metadata:
         }
       ]
     capabilities: Basic Install
-  name: watermarkpodautoscaler.v0.1.0
+  name: watermarkpodautoscaler.v0.2.0-17-g36a0081-dirty
```
Comment (clamoriniere) [inline deploy/olm-catalog/watermarkpodautoscaler/manifests/watermarkpodautoscaler.clusterserviceversion.yaml:42]: I suppose this is not attended

File: deploy/olm-catalog/watermarkpodautoscaler/manifests/watermarkpodautoscaler.clusterserviceversion.yaml [hunk 2]
```diff
@@ -242,4 +242,5 @@ spec:
   maturity: alpha
   provider:
     name: Datadog
-  version: 0.1.0
+  replaces: watermarkpodautoscaler.v0.1.0
+  version: 0.2.0-17-g36a0081-dirty
```
Comment (clamoriniere) [inline deploy/olm-catalog/watermarkpodautoscaler/manifests/watermarkpodautoscaler.clusterserviceversion.yaml:246]: same here

File: deploy/olm-catalog/watermarkpodautoscaler/manifests/watermarkpodautoscaler.clusterserviceversion.yaml (hunk 1 above)
Comment (CharlyF) [reply deploy/olm-catalog/watermarkpodautoscaler/manifests/watermarkpodautoscaler.clusterserviceversion.yaml:42]: yeah, it was added when I generated the olm, I wanted to circle back with Vincent tomorrow to know more

File: pkg/apis/datadoghq/v1alpha1/watermarkpodautoscaler_default.go [hunk 3]
```diff
@@ -5,10 +5,14 @@
...
-import "fmt"
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+)
 
 const (
-	defaultTolerance                       = 0.1
+	defaultTolerance                       = 100
```
Comment (clamoriniere) [inline pkg/apis/datadoghq/v1alpha1/watermarkpodautoscaler_default.go:15]: what is the unit?

Comment (clamoriniere) [reply pkg/apis/datadoghq/v1alpha1/watermarkpodautoscaler_default.go:15]: could you add a comment about this value? what should be the min-max for a tolerance value

File: pkg/apis/datadoghq/v1alpha1/watermarkpodautoscaler_default.go [hunk 4]
```diff
@@ -56,13 +59,14 @@ func IsDefaultWatermarkPodAutoscaler(wpa *WatermarkPodAutoscaler) bool {
 	if wpa.Spec.Algorithm == "" {
 		return false
 	}
-	if wpa.Spec.Tolerance == 0 {
+	// TODO omitempty for the next values, check what an empty resource yields.
```
Comment (clamoriniere) [inline pkg/apis/datadoghq/v1alpha1/watermarkpodautoscaler_default.go:62]: Should we remove this TODO?

File: pkg/controller/watermarkpodautoscaler/replica_calculator.go [hunk 5]
```diff
@@ -177,25 +177,26 @@ func (c *ReplicaCalculator) GetResourceReplicas(logger logr.Logger, target *auto
 
 func getReplicaCount(logger logr.Logger, currentReplicas, currentReadyReplicas int32, wpa *v1alpha1.WatermarkPodAutoscaler, name string, adjustedUsage float64, lowMark, highMark *resource.Quantity) (replicaCount int32, utilization int64) {
 	utilizationQuantity := resource.NewMilliQuantity(int64(adjustedUsage), resource.DecimalSI)
-	adjustedHM := float64(highMark.MilliValue()) + wpa.Spec.Tolerance*float64(highMark.MilliValue())
-	adjustedLM := float64(lowMark.MilliValue()) - wpa.Spec.Tolerance*float64(lowMark.MilliValue())
+	adjustedHM := float64(highMark.MilliValue() + highMark.MilliValue()*wpa.Spec.Tolerance.MilliValue()/1000)
```
Comment (clamoriniere) [inline pkg/controller/watermarkpodautoscaler/replica_calculator.go:180]: should we check first that `wpa.Spec.Tolerance` is not `nil` ?

**Review by clamoriniere — COMMENTED** [review]
few comments about OLM generation, otherwise looks good :)

File: pkg/apis/datadoghq/v1alpha1/watermarkpodautoscaler_default.go (hunk 3 above)
Comment (CharlyF) [reply pkg/apis/datadoghq/v1alpha1/watermarkpodautoscaler_default.go:15]: it's to be used later on as milliquantity, so we can get 10%

File: pkg/controller/watermarkpodautoscaler/replica_calculator.go (hunk 5 above)
Comment (CharlyF) [reply pkg/controller/watermarkpodautoscaler/replica_calculator.go:180]: we will be okay with the defaulting and as it's not a pointer `Tolerance resource.Quantity`
and in `IsDefaultWatermarkPodAutoscaler` we check if the value is == 0, in which cas we default it to 10%.
I added a test case to be explicit about the fact that not specifying the tolerance does not make the spec invalid.
It is however the role of the defaulting methods to ensure we set it to `defaultTolerance`

File: deploy/olm-catalog/watermarkpodautoscaler/manifests/watermarkpodautoscaler.clusterserviceversion.yaml (hunk 1 above)
Comment (vboulineau) [reply deploy/olm-catalog/watermarkpodautoscaler/manifests/watermarkpodautoscaler.clusterserviceversion.yaml]: It's from `VERSION?=$(shell git describe --tags --dirty)` in Makefile. OLM should be generated on tags or we need to change Makefile the way we did it in operator/EDS to take `VERSION` from environment if set.

**Review by vboulineau — APPROVED** [review]

**Pull request merged**